modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/OpenAPIResourceImpl.java

```diff
@@ -53,10 +72,203 @@ public OpenAPIResourceImpl(
+
+		String relatedObjectName = StringUtil.lowerCaseFirstLetter(
+			relatedObjectDefinition.getShortName());
+
+		String customRelationshipEndpoint = StringUtil.replace(
+			endpoint,
+			new String[] {
+				_CURRENT_OBJECT_ID, _RELATIONSHIP_NAME_IN_ENDPOINT,
+				_RELATED_OBJECT_ID
+			},
+			new String[] {
+				currentObjectName + "Id", objectRelationship.getName(),
+				relatedObjectName + "Id"
+			});
+
+		entityPaths.addPathItem(
+			customRelationshipEndpoint, customRelationshipPathItem);
+	}
+
+	private PathItem _createCustomRelationshipPathItem(
+		PathItem relationshipPathItem, ObjectRelationship objectRelationship,
+		ObjectDefinition relatedObjectDefinition) {
+
+		Operation relationshipPutOperation = relationshipPathItem.getPut();
+
+		List<Parameter> relationshipPutOperationParameters =
+			relationshipPutOperation.getParameters();
+
+		Stream<Parameter> relationshipPutOperationParametersStream =
+			relationshipPutOperationParameters.stream();
+
+		List<Parameter> customRelationshipPutOperationParameters =
+			relationshipPutOperationParametersStream.map(
+				parameter -> {
+					Parameter customParameter = new Parameter();
+
+					customParameter.name(parameter.getName());
+					customParameter.in(parameter.getIn());
+					customParameter.required(parameter.getRequired());
+					customParameter.schema(parameter.getSchema());
+
+					return customParameter;
+				}
+			).collect(
+				Collectors.toList()
+			);
+
+		String currentObjectName = StringUtil.lowerCaseFirstLetter(
+			_currentObjectDefinition.getShortName());
+
+		_customizeParameter(
+			customRelationshipPutOperationParameters, _CURRENT_OBJECT_ID,
+			currentObjectName + "Id", false);
+
+		String relatedObjectName = StringUtil.lowerCaseFirstLetter(
+			relatedObjectDefinition.getShortName());
+
+		_customizeParameter(
+			customRelationshipPutOperationParameters, _RELATED_OBJECT_ID,
+			relatedObjectName + "Id", false);
+
+		_customizeParameter(
+			customRelationshipPutOperationParameters, _RELATIONSHIP_NAME,
+			objectRelationship.getName(), true);
+
+		Operation customRelationshipPutOperation = new Operation();
+
+		customRelationshipPutOperation.tags(relationshipPutOperation.getTags());
+		customRelationshipPutOperation.operationId(
+			StringBundler.concat(
+				"put", StringUtil.upperCaseFirstLetter(currentObjectName),
+				StringUtil.upperCaseFirstLetter(objectRelationship.getName()),
+				StringUtil.upperCaseFirstLetter(relatedObjectName)));
+		customRelationshipPutOperation.parameters(
+			customRelationshipPutOperationParameters);
+		customRelationshipPutOperation.responses(
+			relationshipPutOperation.getResponses());
+
+		return new PathItem().put(customRelationshipPutOperation);
+	}
+
+	private void _customizeParameter(
+		List<Parameter> parameters, String parameterName,
+		String newParameterName, boolean shouldBeDeleted) {
+
+		Stream<Parameter> parametersStream = parameters.stream();
+
+		parametersStream.filter(
```

Comment:
When I see this type of Stream.filter predicate I think we might use a Map<String, Parameter> to get the value directly to improve efficiency and readibility